trainers/text_to_image_trainer.py

```diff
@@ -30,16 +33,54 @@ def train_epoch(self, train_loader, epoch_num):
         )
 
         with trange(len(train_loader), **kwargs) as t:
-            for counter, sample in enumerate(train_loader.parsed_subset):
-                # For loop simply to demonstrate the number of images in the batch
-                for i in range(self.batch_size):
-                    image_tensor = np.asarray(Image.open(io.BytesIO(sample['image_raw'].numpy()[i])))
-                    print(counter, image_tensor.shape)
-
-                # image_tensor = np.asarray(Image.open(io.BytesIO(sample['image_raw'].numpy())))
-                # print(counter, image_tensor.shape)
+            for _, sample in enumerate(train_loader.parsed_subset):
+                image_tensor = []
+                text_tensor = []
+                batch_size = len(sample['text'].numpy())
+                for i in range(batch_size):
+                    img = np.asarray(Image.open(io.BytesIO(sample['image_raw'].numpy()[i])), dtype=np.float32)
+                    image_tensor.append(img)
+                    txt = np.frombuffer(sample['text'].numpy()[i], dtype=np.float32).reshape(10, 1024) # TODO make dynamic
+                    text_tensor.append(txt)
+                image_tensor = np.asarray(image_tensor)
+                text_tensor = np.asarray(text_tensor)
                 # For tabular: text_tensor = np.frombuffer(sample['text'].numpy())
                 # For Caption: text_tensor = np.frombuffer(sample['text'].numpy(), dtype=np.float32).reshape(10, 1024)
-                # print(text_tensor)
-                # name = sample['name'].numpy().decode("utf-8")
                 # label = sample['label'].numpy()
+
+                noise_z = tf.random.normal([batch_size, 100])
+
+                with tf.GradientTape() as generator_tape, tf.GradientTape() as discriminator_tape:
+                    smoothed_embedding, mean, log_sigma = self.model.conditional_augmentation(text_tensor)
+                    embedding_z = tf.concat([smoothed_embedding, noise_z], 1)
```

Comment:
Should we be doing this outside the gradient tape maybe?

Reply:
Oh wait never mind, we have some linear layers in there that need to be backpropped.

Reply:
conditional augmentation has the linear layers, so are trainable. Which means they need to be on the tape, no?